### DISCUSSION TITLE
question

### DISCUSSION THREAD

**Discussion by likhita-8091** [filecoin-project/fvm-project#42] [General]
I have a question. The biggest bottleneck of the public chain is performance and ecology. Now most public chains have their own characteristics (playing methods) and some good economic models (unlimited number of additional tokens), such as Polkadot (auction, parallel chain slot, relay chain)), near (shard, nightshade), Solana (proof of history, to achieve great TPS without slice), avalanche EtH 2.0, etc.

They all have similarities. For example, most of them are developed in the rust language, the circulation of tokens is huge, and new tokens, and staking will be issued every year with 10% revenue. What I want to know is that filecoin itself has a fixed circulation of 2 billion. Will the rules be changed in the future? After all, if FVM develops well, there must be less than 2 billion. Moreover, will the consensus mechanism of file coin change and many rules change? In addition, now filecoin also knows whether smart contracts have their own highlights, refreshing, and whether they can solve the performance bottleneck in the industry? If you simply have the smart contract function, it won't help. Maybe it's just good for fil itself.

**Comment by Stebalien** [comment]
The consensus protocol and circulating supply are out of scope in this project. There will have to be some changes to the Filecoin protocol to support the FVM, but that's not going to affect circulating supply, storage, consensus/power, etc.

In terms of performance bottlenecks, we're expecting we'll need sharding at some point but are handling as independent problems.

**Comment by likhita-8091** [comment]
> The consensus protocol and circulating supply are out of scope in this project. There will have to be some changes to the Filecoin protocol to support the FVM, but that's not going to affect circulating supply, storage, consensus/power, etc.
> 
> In terms of performance bottlenecks, we're expecting we'll need sharding at some point but are handling as independent problems.

OK, in fact, FVM is not as great and eye-catching as other public chains. It's just a smart contract.

**Reply by TimGuo7** [reply]
better don't give this statement, it's still under development.

**Reply by likhita-8091** [reply]
> better don't give this statement, it's still under development.

Well, OK, just curious. After all, the blockchain has developed for so many years, and it is not difficult to implement a VM. It is precisely because it is not difficult. If there are no novel points, there is no solution to the performance bottleneck. In fact, this FVM can only be regarded as icing on the cake, not revolutionary innovation

**Reply by raulk** [reply]
@CodingJzy There are many efforts ongoing in the Filecoin ecosystem. The FVM is delivering an execution environment to run custom code on the Filecoin network. Scalability and sharding is being addressed by ConsensusLab. You should look into ConsensusLab and the combination of efforts, instead of judging a single project (FVM) by things that simply not in our scope.
